### ISSUE TITLE
Failed to connect: Access denied for user 'xxx'@'10.46.67.181' (using password: NO) with .my.cnf

### ISSUE THREAD

**Issue by funnng** [rstats-db/RMySQL#120]
im using RMySQL to connect rds(MySQL), it works well with:
`con <- dbConnect(RMySQL::MySQL(), host='rdo112t7w6xn93j.mysql.rds.aliyuncs.com', username='xman', password='xxxxx', dbname='test')`

but failed with .my.cnf:
```
> con <- dbConnect(RMySQL::MySQL(), group = "aliyun-rds")
Error: Failed to connect: Access denied for user 'xman'@'10.46.67.181' (using password: NO)
```
(host rdo112t7w6xn93j.mysql.rds.aliyuncs.com ip is 10.46.67.181)

here is .my.cnf:
```
[aliyun-rds]
host=rdo112t7w6xn93j.mysql.rds.aliyuncs.com
port=3306
username=xman
password=xxxxx
database=test
```

here is session info:
```
> sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

locale:
 [1] LC_CTYPE=en_US.UTF-8          LC_NUMERIC=C                  LC_TIME=en_US.UTF-8          
 [4] LC_COLLATE=en_US.UTF-8        LC_MONETARY=en_US.UTF-8       LC_MESSAGES=en_US.UTF-8      
 [7] LC_PAPER=en_US.UTF-8          LC_NAME=en_US.UTF-8           LC_ADDRESS=en_US.UTF-8       
[10] LC_TELEPHONE=en_US.UTF-8      LC_MEASUREMENT=en_US.UTF-8    LC_IDENTIFICATION=en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RJDBC_0.2-5         rJava_0.9-7         vertica.dplyr_0.4.2 assertthat_0.1      scales_0.3.0       
 [6] ggvis_0.4.2         ggplot2_1.0.1       DBI_0.3.1.9008      DT_0.1              dplyr_0.4.3        
[11] lubridate_1.3.3     shiny_0.12.2.9005  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      magrittr_1.5     MASS_7.3-43      munsell_0.4.2    colorspace_1.2-6 xtable_1.7-4    
 [7] R6_2.1.1         stringr_1.0.0    plyr_1.8.3       tools_3.2.2      parallel_3.2.2   grid_3.2.2      
[13] gtable_0.1.2     htmltools_0.2.6  lazyeval_0.1.10  digest_0.6.8     reshape2_1.4.1   htmlwidgets_0.5 
[19] memoise_0.2.1    mime_0.4         stringi_1.0-1    httpuv_1.3.3     proto_0.3-10   
```

**Comment by funnng** [comment]
also i tried this(change database to dbname) and failed as the same:
```
[aliyun-rds]
host=rdo112t7w6xn93j.mysql.rds.aliyuncs.com
port=3306
username=xman
password=xxxxx
dbname=test
```

**Comment by krlmlr** [comment]
Please try

```
con <- dbConnect(RMySQL::MySQL(), groups = "aliyun-rds")
```

(note the `s` in `groups`). This seems to be necessary due to the ellipsis in the argument list of dbConnect().

Also, the sessionInfo() doesn't include RMySQL.

**Comment by funnng** [comment]
same error
```
> library(DBI)
> library(RMySQL)
> con <- dbConnect(RMySQL::MySQL(), groups = "aliyun-rds")
Error: Failed to connect: Access denied for user 'xman'@'10.46.67.181' (using password: NO)
> traceback()
6: stop(list(message = "Failed to connect: Access denied for user 'xman'@'10.46.67.181' (using password: NO)", 
       call = NULL, cppstack = NULL))
5: .Call("RMySQL_connection_create", PACKAGE = "RMySQL", host, user, 
       password, db, port, unix_socket, client_flag, groups, default_file)
4: connection_create(host, username, password, dbname, port, unix.socket, 
       client.flag, groups, default.file)
3: .local(drv, ...)
2: dbConnect(RMySQL::MySQL(), groups = "aliyun-rds")
1: dbConnect(RMySQL::MySQL(), groups = "aliyun-rds")
> sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RMySQL_0.11.0.9000  RJDBC_0.2-5         rJava_0.9-7         vertica.dplyr_0.4.2 assertthat_0.1     
 [6] scales_0.3.0        ggvis_0.4.2         ggplot2_1.0.1       DBI_0.3.1.9008      DT_0.1             
[11] dplyr_0.4.3         lubridate_1.3.3     shiny_0.12.2.9005  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      magrittr_1.5     MASS_7.3-43      munsell_0.4.2    colorspace_1.2-6 xtable_1.7-4    
 [7] R6_2.1.1         stringr_1.0.0    plyr_1.8.3       tools_3.2.2      parallel_3.2.2   grid_3.2.2      
[13] gtable_0.1.2     htmltools_0.2.6  digest_0.6.8     reshape2_1.4.1   htmlwidgets_0.5  memoise_0.2.1   
[19] mime_0.4         stringi_1.0-1    httpuv_1.3.3     proto_0.3-10    
```

**Comment by krlmlr** [comment]
Looks like this issue is resolved by #90. Could you please install the patch using

```
devtools::install_github("rstats-db/RMySQL#90")
```

and try again?

**Comment by funnng** [comment]
#90 works. thanks for the help
